File: Languages/Prolog/tasklist.pl
:- dynamic tarefa/1.

inicializar :-
    writeln('Listador de Tarefas'),
    menu.

menu :-
    writeln('Escolha uma opção:'),
    writeln('1. Adicionar tarefa'),
    writeln('2. Listar tarefas'),
    writeln('3. Remover tarefa'),
    writeln('4. Sair'),
    read(Op),
    tratar_opcao(Op).

tratar_opcao(1) :-
    adicionar_tarefa,
    menu.
tratar_opcao(2) :-
    listar_tarefas,
    menu.
tratar_opcao(3) :-
    remover_tarefa,
    menu.
tratar_opcao(4) :-
    writeln('Saindo...').
tratar_opcao(_) :-
    writeln('Opção inválida. Tente novamente.'),
    menu.

adicionar_tarefa :-
    writeln('Digite a descrição da tarefa:'),
    read(Tarefa),
    assertz(tarefa(Tarefa)),
    writeln('Tarefa adicionada.').

listar_tarefas :-
    findall(T, tarefa(T), Tarefas),
    (   Tarefas = [] -> writeln('Nenhuma tarefa cadastrada.')
    ;   writeln('Tarefas:'),
        forall(member(T, Tarefas), writeln(T))
    ).

remover_tarefa :-
    writeln('Digite a descrição da tarefa a ser removida:'),
    read(Tarefa),
    (   retract(tarefa(Tarefa)) -> writeln('Tarefa removida.')
    ;   writeln('Tarefa não encontrada.')
    ).

File: Languages/Octave/tasklist.m
function listador_tarefas()
    tarefas = {}; 
    while true
        disp('Listador de Tarefas');
        disp('1. Adicionar tarefa');
        disp('2. Listar tarefas');
        disp('3. Remover tarefa');
        disp('4. Sair');
        opcao = input('Escolha uma opção: ');

        switch opcao
            case 1
                descricao = input('Digite a descrição da tarefa: ', 's');
                tarefas{end + 1} = descricao;  
                disp('Tarefa adicionada.');
            case 2
                if isempty(tarefas)
                    disp('Nenhuma tarefa cadastrada.');
                else
                    disp('Tarefas:');
                    for i = 1:length(tarefas)
                        fprintf('%d: %s\n', i, tarefas{i});
                    end
                end
            case 3
                if isempty(tarefas)
                    disp('Nenhuma tarefa para remover.');
                else
                    listar_tarefas(tarefas);
                    indice = input('Digite o número da tarefa a ser removida: ');
                    if indice >= 1 && indice <= length(tarefas)
                        tarefas(indice) = [];  
                        disp('Tarefa removida.');
                    else
                        disp('Índice inválido.');
                    end
                end
            case 4
                disp('Saindo...');
                return;
            otherwise
                disp('Opção inválida. Tente novamente.');
        end
    end
end

function listar_tarefas(tarefas)
    disp('Tarefas:');
    for i = 1:length(tarefas)
        fprintf('%d: %s\n', i, tarefas{i});
    end
end

listador_tarefas();